docs/data_sharing_guide.md

```diff
@@ -0,0 +1,338 @@
+
+**Auto-Imported Utility Functions:**
+- `check_governance_health()` - Check service status
+- `get_minio_credentials()` - Get your MinIO credentials (sets environment variables)
+- `get_my_sql_warehouse()` - Get your SQL warehouse prefix
+- `get_my_policies()` - Get detailed policy information
+- `get_my_workspace()` - Get comprehensive workspace information
+- `get_path_access_info(path)` - Check who has access to a path
+- `make_table_private(namespace, table_name)` - Remove public access from SQL tables
+- `make_table_public(namespace, table_name)` - Make SQL tables publicly accessible
+- `share_table(namespace, table_name, with_users, with_groups)` - Share SQL tables
+- `unshare_table(namespace, table_name, from_users, from_groups)` - Unshare SQL tables
+
+**Pre-Initialized Client:**
+- `governance` - Pre-initialized `DataGovernanceClient()` instance for advanced operations
+
+**Other Auto-Imported Functions:**
+- `get_spark_session()` - Create Spark sessions with Delta Lake support
+- Plus many other utility functions for data operations
+
+### Quick Start
+
+```python
+# Check your workspace information
+workspace = get_my_workspace()
+print(f"👤 Username: {workspace.username}")
+print(f"🏠 Home paths: {workspace.home_paths}")
+print(f"👥 Groups: {workspace.groups}")
+print(f"📁 Accessible paths: {len(workspace.accessible_paths)}")
+
+# Check service health
+health = check_governance_health()
+print(f"🔍 Service status: {health.status}")
+
+# Get your credentials (automatically sets environment variables)
+credentials = get_minio_credentials()
+print(f"🔐 Credentials loaded for: {credentials.username}")
+```
+
+## Core Information Functions
+
+### Service Health and Credentials
+
+```python
+# Check governance service status
+health = check_governance_health()
+print(f"Service status: {health.status}")
+
+# Get MinIO credentials (automatically sets MINIO_ACCESS_KEY and MINIO_SECRET_KEY)
+credentials = get_minio_credentials()
+print(f"Username: {credentials.username}")
+print(f"Access key: {credentials.access_key}")
+print(f"Secret key: {credentials.secret_key}")
+
+# Get your SQL warehouse prefix
+sql_warehouse = get_my_sql_warehouse()
+print(f"SQL warehouse prefix: {sql_warehouse.sql_warehouse_prefix}")
+
+# The pre-initialized governance client is also available
+print(f"Governance client ready: {governance is not None}")
+```
+
+### Workspace and Policy Information
+
+```python
+# Get comprehensive workspace information
+workspace = get_my_workspace()
+print(f"Username: {workspace.username}")
+print(f"Home paths: {workspace.home_paths}")
+print(f"Groups: {workspace.groups}")
+print(f"Total accessible paths: {len(workspace.accessible_paths)}")
+
+# Get detailed policy information
+policies = get_my_policies()
+print(f"Home policy: {policies.user_home_policy.policy_name}")
+print(f"System policy: {policies.user_system_policy.policy_name}")
+print(f"Group policies: {len(policies.group_policies)}")
```

Comment:
IIUC the policy data structures are relatively complex and may need some explanation

Reply:
I'm not sure how valuable it is to explain the policy document here. This method is primarily intended for developers to troubleshoot issues, and I doubt most users would be interested in reading through the policy details.

Reply:
Ok. If they're not intended for users I'd remove the policy doc examples in that case, and possibly put them in some developer oriented docs

Reply:
👍 removed from guide

Reply:
:+1: